### ISSUE TITLE
Error when running tests with v2.11 

### ISSUE THREAD

**Issue by gabrielgrant** [ember-cli/ember-cli#6723]
I began seeing the following error in my test output immediately after updating my global CLI to 2.11 (ie before i'd actually updated my project):

```
global failure (1, 0, 1)
Uncaught SyntaxError: Unexpected token o in JSON at position 1@ 6 ms
Source: 	
http://localhost:7357/1090/tests/index.html?hidepassed:1
Source: at window.onerror (http://localhost:7357/assets/test-support.js:4457:7)
```

The console contains the following two errors: 

```
VM1874:1 Uncaught SyntaxError: Unexpected token o in JSON at position 1
deserializeMessage	@	testem.js:881
messageListener	@	testem.js:849
```
and
```
Failed to parse SourceMap: http://localhost:7357/socket.io/socket.io.js.map
```

Unfortunately updating the project to 2.11 doesn't fix the issue. Other than the error appearing when running tests, the app seems to function properly.

Error was confirmed by `@liquid-swordsman` in [the testing slack channel also after upgrade to v2.11](https://embercommunity.slack.com/archives/-testing/p1485556814004305)

---

Output from `ember version --verbose && npm --version`:
```
ember-cli: 2.11.0
http_parser: 2.7.0
node: 6.3.0
v8: 5.0.71.52
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 57.1
modules: 48
openssl: 1.0.2h
os: linux x64
3.10.3

```

Tested using Chromium Version 53.0.2785.143

**Comment by nathanhammond** [comment]
We can't really investigate this without a reproduction. Can you create a repository that demonstrates this bug and push it to GitHub? (`ember new`, work to recreate)

**Comment by gabrielgrant** [comment]
Thanks for the quick reply -- was already working on it :)

**Comment by gabrielgrant** [comment]
MVCE here: https://github.com/gabrielgrant/ember-2.11-test-error

FWIW, just opening the testsuite created by the ember-cli v2.11 default blueprint is enough to trigger the error.

**Comment by nathanhammond** [comment]
@gabrielgrant I cannot reproduce this. Can you reinstall your node modules?

**Comment by gabrielgrant** [comment]
you mean rm the `node_modules` dir in that project and then run `npm install` again?

**Comment by nathanhammond** [comment]
`git clean -fdx`

**Comment by gabrielgrant** [comment]
k, did that (and ran `npm install` after -- `ember test --serve` wouldn't run without that), but still getting the same error

**Comment by gabrielgrant** [comment]
Actually, seems to be slightly different this time: there are two copies of the JSON error in the console:

```
VM2907:1 Uncaught SyntaxError: Unexpected token o in JSON at position 1
deserializeMessage	@	testem.js:881
messageListener	@	testem.js:849
VM2917:1 Uncaught SyntaxError: Unexpected token o in JSON at position 1
deserializeMessage	@	testem_connection.js:23
handleMessage	@	testem_connection.js:47
```

**Comment by gabrielgrant** [comment]
To be clear, the direct cause of these errors are trying to parse a stringified JS object (`[object Object]`) as JSON. But it's not clear to me why testem is trying to do that (the messages coming into `messageListener` and `handleMessage` must be coming from somewhere, but it's not clear where (and the `[object Object]` string representation doesn't exactly help figure out the source)

**Comment by tomdale** [comment]
I'm hitting this too. In my case, I believe it was "caused" by the Chrome Accessibility Developer Tools triggering a non-string message on `window`. However, Testem should be resilient to unknown messages—currently, it's pooping in the shared `postMessage` pool everyone needs to use.

**Comment by marclipovsky** [comment]
This still seems to be happening with Ember 2.14.0. Any updates?

![screen shot 2017-07-12 at 5 55 43 pm](https://user-images.githubusercontent.com/112484/28145984-55b4efa8-672b-11e7-94f3-db47758f026d.png)

**Comment by tomdale** [comment]
@marclipovsky I couldn't find a bug filed with testem, would you mind filing one? Per my [earlier comment](https://github.com/ember-cli/ember-cli/issues/6723#issuecomment-286299900) it seems like testem blindly uses any messages send over `postMessage` without doing any data validation. In my case, a Chrome extension like the Accessibility Developer Tools was sending a message testem couldn't understand and crashing.

As a workaround you might try running the tests in a Chrome profile with extensions disabled.

**Comment by marclipovsky** [comment]
Ah my bad, misunderstood. Yes I just added the `--extensions-disabled` flag to my `browser_args` and don't see the issue anymore. Thanks @tomdale!

**Comment by trentmwillis** [comment]
I've opened an issue with Testem (https://github.com/testem/testem/issues/1156) and will try to submit a fix when I have time (though if someone beats me to it, that'd be great). Unsure if this issue needs to remain open in the interim.

**Comment by tomdale** [comment]
@trentmwillis Thanks Trent! Will close this issue and hope this gets patched in Testem.

**Comment by trentmwillis** [comment]
Closing the loop here: a fix has been merged + released in `testem@1.18.2`
